### DISCUSSION TITLE
Do folks need/want OTLP support?

### DISCUSSION THREAD

**Discussion by schmidtw** [xmidt-org/xmidt#69] [General]
The [candlelight](https://github.com/xmidt-org/candlelight) library support Zipkin, Jaeger and stdout, but not OTLP.  Are folks finding this to be an issue we should work to address?

**Comment by ctorrao** [comment]
Currently we are investing in having a full OpenTelemetry architecture using the vendor-agnostic approach. Therefore, we interconnect all telemetry that outputs from our application services to an intermediate collector using OLTP protocol.

We only output for Prometheus, Zipkin or Jaeger in that last mile using a central OpenTelemetry collector (exporter feature). This way, we ensure that all application services do not need to know each protocol/format from each vendor or UIs/dashboards/frontends.

This example images from the OpenTelemetry site somehow show the approach I was referring previously:
![image](https://user-images.githubusercontent.com/22594902/186623795-a77ebcac-d90a-4b19-a1e6-15b9112794fb.png)
![image](https://user-images.githubusercontent.com/22594902/186624116-67840420-9a02-46f9-82f4-eaffc98f6a08.png)

Regarding to the change, if it makes sense to integrate that in the common repo, I can check internally if someone from my side can develop and open a PR with the change (it depends a lot if it is something easy to enter in the codebase or not). Is ok for you this feature?

Refs:
https://opentelemetry.io/docs/
https://opentelemetry.io/docs/collector/

**Comment by ctorrao** [comment]
I was checking your email, and regarding the OLTP feature set, I can ensure you that the protocol already supports Traces & Metrics (we tested already that in some pilot microservices). The only part that it's still in development is the logging.

For the GO SDK, this is the current status (the Metrics part it's still in Alpha):
![image](https://user-images.githubusercontent.com/22594902/186627904-89472009-9b06-45f9-afca-7fc8b56c47c5.png)

Ref: https://opentelemetry.io/docs/instrumentation/go/#status-and-releases

**Comment by schmidtw** [comment]
If I'm understanding correctly, we just need to add OTLP support in candlelight & that should allow configuring the servers to point to an OTLP collector.  We might be able to work from https://github.com/xmidt-org/candlelight/pull/33 for a start at least.

I'm in favor of adding what we think will help us use the tooling better.  What I'm not sure of is if this is enough or if there are other things we need to do too.

**Comment by emanuelef** [comment]
Hi, just wanted to ask what's the status of OTel support. I see that mux seems to be instrumented but haven't found any documentation on how to enable instrumentation. Is it just adding the standard OTEL env vars like OTEL_EXPORTER_OTLP_ENDPOINT ?
